ethereum/client/src/tests.rs

```diff
@@ -137,7 +135,68 @@ where
         message_queue_proxy: *message_queue_proxy.address(),
         relayer_proxy: *relayer_proxy.address(),
         erc20_treasury_proxy: *erc20_treasury_proxy.address(),
-    })
+    };
+
+    let nonce = provider
+        .get_transaction_count(signer_address)
+        .await
+        .map_err(|_| Error::ErrorInHTTPTransport)?;
+
+    let relayer_proxy = ProxyContract::new(deployment.relayer_proxy, provider.clone());
+
+    let pending_tx = relayer_proxy
+        .upgradeToAndCall(deployment.relayer, Bytes::new())
+        .nonce(nonce);
+
+    let pending_tx = pending_tx
+        .send()
+        .await
+        .map_err(Error::ErrorDuringContractExecution)?;
+
+    pending_tx
+        .get_receipt()
+        .await
+        .map_err(|_| Error::ErrorWaitingTransactionReceipt)?;
+
+    let nonce = provider
+        .get_transaction_count(signer_address)
+        .await
+        .map_err(|_| Error::ErrorInHTTPTransport)?;
```

Comment:
Move it to separate `fn`

Reply:
And also - can `nonce` be managed by provider itself?

Reply:
Let's wait for alloy fix : https://github.com/alloy-rs/alloy/issues/1073